pyvista/core/filters/image_data.py

```diff
@@ -424,13 +423,12 @@ def image_threshold(
         else:
             field = self.get_array_association(scalars, preference=preference)
 
-        # For some systems and/or Numpy < 2.0, integer scalars won't threshold
-        # correctly. Cast to float in these cases.
-        has_int_dtype = np.issubdtype(
+        # For some systems integer scalars won't threshold
+        # correctly. Cast to float in these cases to be robust.
```

Comment:
Since we no longer have a platform check:
```suggestion
        # correctly. Cast to float to be robust.
```